tiny_dnn/nodes.h

```diff
@@ -43,24 +43,32 @@ namespace cereal {
 
 template <typename Archive>
 void save(Archive & ar, const std::vector<tiny_dnn::layerptr_t>& v) {
+#ifndef CNN_NO_SERIALIZATION
     ar(cereal::make_size_tag((cereal::size_type)v.size()));
     for (auto n : v) {
         tiny_dnn::layer::save_layer(ar, *n);
     }
+#else
+    throw tiny_dnn::nn_error("TinyDNN was not build with Serialization support");
```

Comment:
By moving `cereal` to `third_party` I remember from @nyanp that the issue was about the include paths.
Regarding serialization, I'd move codes to `tiny_dnn/core/serialzation` or  `tiny_dnn/core/util/serialzation` instead.